bin/action_runner.py

```diff
@@ -116,30 +115,43 @@ def run_command(command):
     )
 
 
-def stdout_reader(proc):
-    for line in iter(proc.stdout.readline, b''):
-        sys.stdout.write(line.decode('utf-8'))
-        sys.stdout.flush()
+def stream(source, dst):
+    is_connected = True
+    logging.warning(f'streaming {source} to {dst}')
+    for line in iter(source.readline, b''):
+        if is_connected:
+            try:
+                dst.write(line.decode('utf-8'))
+                dst.flush()
+                logging.warning(line)
+            except Exception as e:
+                logging.warning(f'failed writing to {dst}: {e}')
+                logging.warning(line)
+                is_connected = False
+        else:
+            logging.warning(line)
+            is_connected = False
 
 
-def stderr_reader(proc):
-    for line in iter(proc.stderr.readline, b''):
-        sys.stderr.write(line.decode('utf-8'))
-        sys.stdout.flush()
+def configure_logging(run_id):
+    logging.basicConfig(filename=f'/tmp/{run_id}.log')
```

Comment:
`/tmp/action_runner.<pid>.<run_id>.log`?

Reply:
hm. we've not got the pid at this point, so can't do this.